astropy/io/fits/src/compressionmodule.c

```diff
@@ -233,159 +235,136 @@ int compress_type_from_string(char* zcmptype) {
 }
 
 
+PyObject *
+get_header_value(PyObject* header, char* key) {
+    PyObject* hdrkey;
+    PyObject* hdrval;
+    hdrkey = PyUnicode_FromString(key);
+    if (hdrkey == NULL) {
+        return NULL;
+    }
+    hdrval = PyObject_GetItem(header, hdrkey);
+    Py_DECREF(hdrkey);
+    PyErr_Clear();
+    return hdrval;
+}
+
+
 // TODO: It might be possible to simplify these further by making the
 // conversion function (eg. PyString_AsString) an argument to a macro or
 // something, but I'm not sure yet how easy it is to generalize the error
 // handling
+/* The get_header_* functions resemble "Header.get" where "def" is the default
+   value, "keyword" is a string representing the header-key and the result is
+   stored in "val".
+   The function returns 0 on success, 1 if the header didn't have the keyword
+   and the default was applied and -1 (with an exception set) if an Exception
+   happened (like a MemoryError or Overflow).
+*/
 int get_header_string(PyObject* header, char* keyword, char* val, char* def) {
-    PyObject* keystr;
-    PyObject* keyval;
-    PyObject* tmp;  // Temp pointer to decoded bytes object
-    int retval;
-
-    keystr = PyUnicode_FromString(keyword);
-    keyval = PyObject_GetItem(header, keystr);
+    PyObject* keyval = get_header_value(header, keyword);
 
     if (keyval != NULL) {
```

Comment:
Just a minor stylistic comment, but if you handle the  `keyval == NULL` case first and have an early return, then the code below does not have to be in an enclosed block.